sdcm/kafka/kafka_consumer.py

```diff
@@ -0,0 +1,107 @@
+
+
+class KafkaCDCReaderThread(Thread):  # pylint: disable=too-many-instance-attributes
+    """
+    thread that listen on kafka topic, and list all the unique key
+    received, so we can validate how many unique key we got
+    """
+
+    def __init__(self, tester, params: SCTConfiguration, kafka_addresses: list | None = None,  # pylint: disable=too-many-arguments
+                 connector_index: int = 0, group_id: str = None, duration: int | None = None, **kwargs):
+        self.keys = set()
+        self.termination_event = Event()
+        self.params = params
+        self.tester = tester
+        self.duration = duration
+        self._kafka_addresses = kafka_addresses
+        self.group_id = group_id or generate_random_string(16)
+        self.read_number_of_key = int(kwargs.get('read_number_of_key', 0))
+
+        connector_config: SctKafkaConfiguration = params.get("kafka_connectors")[connector_index]
+
+        # TODO: handle setup of multiple tables
+        topic = f'{connector_config.config.scylla_name}.{connector_config.config.scylla_table_names}'
+        self.wait_for_topic(topic, timeout=60)
+        self.consumer = kafka.KafkaConsumer(
+            topic,
+            auto_offset_reset='earliest',
+            enable_auto_commit=True,
+            auto_commit_interval_ms=1000,
+            group_id=self.group_id,
+            bootstrap_servers=self.kafka_addresses,
+        )
+
+        super().__init__(daemon=True)
+
+    @property
+    def kafka_addresses(self):
+        if self.params.get('kafka_backend') == 'localstack':
+            return ['localhost']
+        elif self._kafka_addresses:
+            return self._kafka_addresses
+        return None
+
+    def get_topics(self):
+        admin_client = kafka.KafkaAdminClient(bootstrap_servers=self.kafka_addresses)
+        topics = admin_client.list_topics()
+        LOGGER.debug(topics)
+        return topics
+
+    def wait_for_topic(self, topic, timeout):
+        def check_topic_exists():
+            topics = self.get_topics()
+            return topic in topics
+
+        wait_for(check_topic_exists, text=f"waiting for topic={topic}", timeout=timeout)
+
+    def run(self):
+        while not self.termination_event.is_set():
+            records = self.consumer.poll(timeout_ms=1000)
+            for _, consumer_records in records.items():
+                for msg in consumer_records:
+                    data = json.loads(msg.value).get('payload', {}).get('after', {})
+                    key = base64.b64decode(data.get('key')).decode()
+                    self.keys.add(key)
+                    LOGGER.debug(key)
+            if len(self.keys) >= self.read_number_of_key:
+                self.stop()
+
+    def stop(self):
+        self.termination_event.set()
+        self.consumer.close()
+
+    def kill(self):
+        self.termination_event.set()
+        self.consumer.close()
+
+    def verify_results(self) -> (list[dict | None], list[str | None]):
```

Comment:
not used, not implemented

Reply:
it is called, since we use this thread as a stress command.